alwaysontop/main.js

```diff
@@ -62,13 +70,56 @@ function onAlwaysOnTopWindow(
 }
 
 /**
- * Returns the coordinates for the always on top window in order to display it
- * in the top right corner.
+ * Calculates the coordinates for a window (based on its current coordinates and
+ * size) to place it in the boundaries of a given screen.
+ *
+ * @param {Rectangle} windowRectangle - The current position and dimensions of
+ * the window.
+ * @param {Rectangle} screenRectangle - The position and dimensions of the
+ * screen.
+ * @returns {Point} - The new coordinates for the window.
+ */
+function positionWindowWithinScreenBoundaries(
+        windowRectangle,
+        screenRectangle) {
+    const maxY
+        = screenRectangle.y + screenRectangle.height - windowRectangle.height;
+    const maxX
+        = screenRectangle.x + screenRectangle.width - windowRectangle.width;
+
+    return {
+        x: Math.min(Math.max(windowRectangle.x, screenRectangle.x), maxX),
+        y: Math.min(Math.max(windowRectangle.y, screenRectangle.y), maxY)
+    };
+}
+
+/**
+ * Returns the stored coordinates for the always on top window of its previous
+ * location or if the last location is unknown returns coordinates to display
+ * the window in the top right corner of the screen.
```

Comment:
When I tested I was seeing it in the top left. Is there an issue with the logic or the comment?

Reply:
it should be the logic!